src/openapi/streaming/connection/transport/websocket-transport.ts

```diff
@@ -526,57 +544,51 @@ class WebsocketTransport implements StreamingTransportInterface {
         };
         const url = `${this.authorizeUrl}?contextId=${contextId}`;
 
-        this.authorizePromise = fetch('put', url, options)
-            .then((response) => {
+        fetch('put', url, options).then(
+            () => {
                 log.debug(LOG_AREA, 'Authorization completed', {
                     contextId,
                 });
-                return response;

```

Comment:
The core of the issue here is that there is a single promise assigned to the class, but there can be multiple calls in progress. The start command might listen to `then` on a promise that does not complete

Reply:
instead of getting a promise, you call a function to do something and there is a call to onAuthorised when it is done.